src/benchmark/file_io_write_mmap_benchmark.cpp

```diff
@@ -0,0 +1,114 @@
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <algorithm>
+#include <numeric>
+
+#include "file_io_write_micro_benchmark.hpp"
+
+namespace hyrise {
+
+class FileIOWriteMmapBenchmarkFixture : public FileIOWriteMicroBenchmarkFixture {
+ protected:
+  void mmap_write_benchmark(benchmark::State& state, const int flag, const int data_access_mode, const ssize_t file_size);
+};
+
+/*
+ * Performs a benchmark run with the given parameters. 
+ * 
+ * @arguments:
+ *      state: the benchmark::State object handed to the called benchmarking function.
+ *      flag: The mmap flag (e.g., MAP_PRIVATE or MAP_SHARED).
+ *      data_access_mode: The way the data is written.
+ *                  (-1)  No data access
+ *                  (0)   Sequential
+ *                  (1)   Random
+ *      file_size: Size argument of benchmark.
+*/
+void FileIOWriteMmapBenchmarkFixture::mmap_write_benchmark(benchmark::State& state, const int flag,
+                                                            const int data_access_mode, const ssize_t file_size) {
+  auto fd = int32_t{};
+  if ((fd = open(filename, O_RDWR)) < 0) {
+    close(fd);
+    Fail("Open error:" + std::strerror(errno));
+  }
```

Comment:
In the future we might refactor this to an Assert with a function call? f.e.
```suggestion
  Assert((fd = open(filename, O_RDWR)) < 0,  my_fail(fd, errno));
```

Reply:
something like `fail_and_close_file(..)` perhaps? would be prettier

Reply:
has been done